enhancements/accelerators/out-of-tree-drivers.md

```diff
@@ -0,0 +1,418 @@
+
+
+
+### kmods-via-containers (KVC)
+
+kmods-via-containers is a framework for building and delivering kernel modules via containers. This implementation for this framework was inspired by the work done by Joe Doss on atomic-wireguard.
+This framework relies on 3 independently developed pieces.
+
+1. The kmods-via-containers code/config (this repo)
+
+Delivers the stencil code and configuration files for building and delivering kmods via containers. It also delivers a service kmods-via-containers@.service that can be instantiated for each instance of the KVC framework.
+
+2. The kernel module code that needs to be compiled
+
+This repo represents the kernel module code that contains the source code for building the kernel module. This repo can be delivered by vendors and generally nothing about containers. Most importantly, if someone wanted to deliver this kernel module via the KVC framework, the owners of the code don't need to be consulted.
+
+3. A KVC framework repo for the kernel module to be delivered
+
+This repo defines a container build configuration as well as a library, userspace tools, and config files that need to be created on the host system. This repo does not have to be developed by the owner of the kernel module that is wanted to be delivered.
+
+It must define a few functions in the bash library:
+
+* build_kmods()
+  * Performs the kernel module container build
+* load_kmods()
+  * Loads the kernel module(s)
+* unload_kmods()
+  * Unloads the kernel module(s)
+* wrapper()
+  * A wrapper function for userspace utilities
+
+Customers can hook in their procedures on how to build, load, unload etc their kernel modules. We are providing only an interface not the actual "complicated" implementation of those steps (facade pattern). Customers can then use any tool(s) (akmods, dkms, ..) they need to build their modules. 
+
+Additionally to those callbacks/hooks we need functionality that prepares the kmods (copy the artifacts) and proxy setup. The package managers are ignoring the environment variables for proxies, an extra step is needed to make a container fully proxy aware. 
+
+## Motivation
+
+In OpenShift v3.x out-of-tree drivers could be easily installed on the nodes, since the node was a full RHEL node with a subscription and needed tools were installed easily with yum. 
+In OpenShift v4.x this changed with the introduction of RHCOS. There are currently two different documented ways to enable out-of-tree drivers. One way is using a SRO based operator such as the nvidia GPU operator and the other is using kmods-via-containers. 
+We want to come up with a unified solution that works for our customers across RHCOS and RHEL. The solution should also help customers that are currently on RHEL7 to consider moving to RHCOS which is fully managed and easier to support in OpenShift.
+ 
+#### Day-1 DriverContainer Management  [OpenShift | BareMetal]
+ 
+TODO: LONGER TERM
```

Comment:
I _think_ kmods-via-containers fits in pretty well here. While it lacks the tighter integration with the cluster it does have the ability to execute builds and inject the kmod way before the cluster is ready to go.